proof_of_play.go

```diff
@@ -129,10 +131,21 @@ func (p proofOfPlay) publishEvent(name string, message string, level string) {
 	p.eventFn(name, message, "", level)
 }
 
-func (p *proofOfPlay) processResponse(popType string, adId string,
-	resp *http.Response) error {
-	code := resp.StatusCode
+func (p *proofOfPlay) makePoPRequest(req *http.Request,
+	popReq *PoPRequest, adId string, popType string) error {
+	resp, err := p.httpClient.Do(req)
+	if err != nil {
+		// Connection error - retry the request
+		p.processRequestFailure(popReq, err)
+		return &PoPError{
+			Status:  http.StatusAccepted,
+			Message: fmt.Sprintf("Connection Error: %s", err.Error()),
```

Comment:
we should tell the user that the request will be retried, perhaps something like: `Connection error, request will be retried: %s`